crates/core/tedge_api/src/alarm.rs

```diff
@@ -31,6 +31,8 @@ pub enum AlarmSeverity {
 pub struct ThinEdgeAlarmData {
     pub text: Option<String>,
 
+    pub message: Option<String>,
```

Comment:
I would definitely not add a `message` field along the `text` field, both having the very same role.

* This is an issue related to C8Y and should only affect the `C8yCreateAlarm` struct.
* It's already confusing to deal with both  `text` and `message` in the mapping for C8Y. One must not move this confusion to thin-edge itself.

Reply:
I am confused here because the spec <https://github.com/thin-edge/thin-edge.io-specs/blob/main/src/alarms/alarm-handling.md> says that, the `message` will not be part of the `alarm`. But the document <https://github.com/thin-edge/thin-edge.io/blob/main/docs/src/tutorials/raise-alarm.md> says that `text` can be part of the alarm message. So, need some clarification.

Reply:
Yes we have a mix of the two fragments throughout the documentation/spec.

The `.message` fragment makes more sense to me as it promotes a more agnostic model. So the `.message` should be the official fragment which is part of the tedge alarm model. In that context, `.text` should just be considered as any other custom fragment. It is only when it comes to the c8y-mapper that it needs to reconcile the `.text` and `.message` fragments (as per the handling detailed in the ticket).

Reply:
However if the `.message` parsing was never implemented, then I would also be open to making `.text` part of the official tedge alarm schema (that would simplify the logic for the c8y-mapper).

Reply:
If `message` makes more sense, then we can keep only that one. I only want to avoid two fields with the purpose.

Reply:
Yes I completely agree. The map is there to be a catch all for the other fragments.

Reply:
So, in that case, the scheme for the `alarm` message is `{"message":"msg","time":"datetime", ///rest s custom fragment}`. Map the `message` to `text` in case of a c8y alarm message. If there is a `text` field in the input alarm message then it will go as a custom fragment.

Reply:
We had another discussion and we have chosen to take a different approach (Option 2). Option 2, involves changing the specification to say that the alarm description should be provided in the `.text` fragment. This way any other fragments are just treated as custom fragments which tedge handles transparently.

More details/reasons are provided in this comment:
https://github.com/thin-edge/thin-edge.io/issues/1710#issuecomment-1423706337